nvdaHelper/local/UIAEventLimiter/rateLimitedEventHandler.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+This file is a part of the NVDA project.
+URL: http://github.com/nvaccess/nvda/
+Copyright 2023 NV Access Limited.
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 2.0, as published by
+    the Free Software Foundation.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+This license can be found at:
+http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+*/
+
+#include <windows.h>
+#include <uiautomation.h>
+#include <numeric>
+#include <variant>
+#include <map>
+#include <vector>
+#include <functional>
+#include <mutex>
+#include <atomic>
+#include <atlcomcli.h>
+#include <comutil.h>
+#include <common/log.h>
+#include "eventRecord.h"
+#include "rateLimitedEventHandler.h"
+
+template<EventRecordConstraints EventRecordClass, typename... EventRecordArgTypes>
+HRESULT RateLimitedEventHandler::queueEvent(EventRecordArgTypes&&... args) {
+	LOG_DEBUG(L"RateLimitedUIAEventHandler::queueEvent called");
+	bool needsFlush = false;
+	{ std::lock_guard lock(m_mtx);
```

Comment:
should this be on a new line?

Reply:
I can... Though I personally like this style specifically for scoped locks as it makes it very clear the scope is for that purpose. If it is confusing we can add a new line.

Reply:
FWIW, here's another style I've seen to deal with this:

```
{ // Scope for lock
  std::lock_guard lock(m_mtx);
```

This makes it clear what the block is for without potentially making it look like a misplaced brace.

Reply:
Yep, I'm personally happy with that style.

Reply:
```suggestion
	{ // scoped lock
		std::lock_guard lock(m_mtx);
```